src/jaxsim/api/data.py

```diff
@@ -819,9 +833,14 @@ def random_model_data(
             key=k1, shape=(3,), minval=p_min, maxval=p_max
         )
 
-        physics_model_state.base_quaternion = jaxlie.SO3.from_rpy_radians(
-            *jax.random.uniform(key=k2, shape=(3,), minval=0, maxval=2 * jnp.pi)
-        ).wxyz
+        physics_model_state.base_quaternion = jaxsim.math.Quaternion.to_wxyz(
+            xyzw=jax.scipy.spatial.transform.Rotation.from_euler(
+                seq=base_rpy_seq,
+                angles=jax.random.uniform(
+                    key=k2, shape=(3,), minval=rpy_min, maxval=rpy_max
+                ),
+            ).as_quat()
+        )
```

Comment:
Maybe we should add `scalar_first=True` in order to set the quaternion as **wxyz** 
```suggestion
        physics_model_state.base_quaternion = jax.scipy.spatial.transform.Rotation.from_euler(
            seq=base_rpy_seq,
            angles=jax.random.uniform(
                key=k2, shape=(3,), minval=rpy_min, maxval=rpy_max
            ),
        ).as_quat(scalar_first=True)
```

Reply:
The JAX implementation, at the time of writing, does not have `scalar_first` as in official SciPy:

https://github.com/google/jax/blob/427a490d2b4147c41c84560792936d693d3c921a/jax/_src/scipy/spatial/transform.py#L170-L175

Reply:
Oh alright, then what about something like:

```python
Q = jax.scipy.spatial.transform.Rotation.from_euler(
            seq=base_rpy_seq,
            angles=jax.random.uniform(
                key=k2, shape=(3,), minval=rpy_min, maxval=rpy_max
            ),
        ).as_quat()
Q = jnp.roll(Q, shift=-1)
```

Reply:
When I read the roll code, to me is never clear the direction of rolling. I prefer much more the more verbose call `Quaternion.to_wxyz(xyzw=...)` that at least is explicit on what it is doing. Considering the runtime, there should not be any difference.

Reply:
Alright, sounds good